service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java

```diff
@@ -230,17 +232,40 @@ void shouldReturnMostRecentUploadedOrderDocumentUrl() {
             returnedMostRecentUploadedOrderDocumentUrl);
     }
 
-    private void assertCommonC21Fields(GeneratedOrder order) {
-        assertThat(order.getType()).isEqualTo(BLANK_ORDER);
-        assertThat(order.getDocument()).isEqualTo(DocumentReference.builder().build());
-        assertThat(order.getDetails()).isEqualTo("Some details");
-        assertThat(order.getDate()).isNotNull();
-        assertThat(order.getJudgeAndLegalAdvisor()).isEqualTo(JudgeAndLegalAdvisor.builder().build());
+    @Test
+    void shouldRemoveOrderPropertiesWhenTheyExistInCaseDetails() {
+        Map<String, Object> data = Arrays.stream(GeneratedOrderKey.values())
+            .collect(Collectors.toMap(GeneratedOrderKey::getKey, value -> ""));
+
+        data.put("DO NOT REMOVE", "");
+
+        service.removeOrderProperties(data);
+
+        assertThat(data).containsOnlyKeys("DO NOT REMOVE");
+    }
+
+    private static Stream<Arguments> fileNameSource() {
+        return Stream.of(
+            Arguments.of(BLANK_ORDER, null, "blank_order_c21.pdf"),
+            Arguments.of(CARE_ORDER, INTERIM, "interim_care_order.pdf"),
+            Arguments.of(CARE_ORDER, FINAL, "final_care_order.pdf"),
+            Arguments.of(CARE_ORDER, null, "care_order.pdf")
+        );
+    }
+
+    private static Stream<Arguments> docmosisDataGenerationSource() {
+        return Stream.of(
+            Arguments.of(BLANK_ORDER, null),
+            Arguments.of(CARE_ORDER, INTERIM),
+            Arguments.of(CARE_ORDER, FINAL)
+        );
     }
 
-    @SuppressWarnings("unchecked")
-    private Map<String, Object> createExpectedOrderData(String date, GeneratedOrderType type) {
-        ImmutableMap.Builder expectedMap = ImmutableMap.<String, Object>builder();
+    private Map<String, Object> createExpectedDocmosisData(GeneratedOrderType type,
+                                                           GeneratedOrderSubtype subtype,
+                                                           LocalDateTime date) {
+        ImmutableMap.Builder<String, Object> expectedMap = ImmutableMap.builder();
+        String formattedDate = dateFormatterService.formatLocalDateToString(date.toLocalDate(), FormatStyle.LONG);
 
         switch (type) {
```

Comment:
As agreed, we can get rid of the switch statement here.

Reply:
I think we should leave these switch states so that we can use the parameterised test 'shouldCreateExpectedMapWhenGivenPopulatedCaseData()' nicely. The file is long enough and would get even longer if we have to split the switch out into separate private methods (and then pass different caseDatas as our parameters instead)